### PR TITLE
Serve Image Files

### PR DESCRIPTION
**Pull request by codeidoscope**
- [x] Serve image files
- [x] Accidentally, also serve PDF files
- [x] Refactor `Response` object to take `Header` and `Body` types
- [x] Get MIME type from file to dynamically generate header
- [x] Create `HeaderGenerator` to automate header creation

### CODE REVIEW

File: src/main/java/com/github/codeidoscope/DefaultHandler.java [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
-import java.nio.charset.StandardCharsets;
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 
 class DefaultHandler implements RouteHandler {
     @Override
     public Response respondToRequest(Request request) {
-        String body = "404 - NOT FOUND";
+        Body body = new Body("404 - NOT FOUND".getBytes());
```
Comment (junglie85) [inline src/main/java/com/github/codeidoscope/DefaultHandler.java:9]: Could you encapsulate the `toBytes()` message within the constructor? Also for `Header`?

File: src/main/java/com/github/codeidoscope/DirectoryHandler.java [hunk 2]
```diff
@@ -2,27 +2,21 @@
...
         String filePath = contentRootPath + request.getPath();
 
-        String body = generateBodyFromDirectory(filePath);
-
-        String statusCode = "200 OK";
-        String date = DateTimeFormatter.RFC_1123_DATE_TIME.format(ZonedDateTime.now());
-        String contentLength = "" + body.getBytes(StandardCharsets.UTF_8).length;
-        String contentType = "text/html";
-        String headers = request.getProtocol() + " " + statusCode + "\n" + "Date: " + date + "\n" + "Content-Type: " + contentType + "\n" + "Content-Length: " + contentLength;
+        Body body = new Body(generateBodyFromDirectory(filePath).getBytes());
```
Comment (junglie85) [inline src/main/java/com/github/codeidoscope/DirectoryHandler.java:16]: Could this `toBytes()` be encapsulated in the constructor?

File: src/test/java/com/github/codeidoscope/HeaderGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class HeaderGeneratorTest {
+
+    @Test
+    void aNewHeaderObjectWithCorrectDataIsCreated() {
+        HeaderGenerator headerGenerator = new MockHeaderGenerator();
```
Comment (junglie85) [inline src/test/java/com/github/codeidoscope/HeaderGeneratorTest.java:11]: Did you mean to test the mock header generator, or did you intend to test the request header generator?

**Comment by junglie85** [comment]
Looks good @codeidoscope. I like the introduction of the `HeaderGenerator` :+1:.

**Comment by codeidoscope** [comment]
@AshleyByeUK For posterity, the fix was made on this branch 'cause I'd already merged this one to master 👀: https://github.com/codeidoscope/aguafresca/pull/4/commits/0c4552b15c646f2a84a0f471fbd895739280ab20

**Pull request merged**